sound/soc/intel/boards/sof_sdw.c

```diff
@@ -347,7 +485,7 @@ static int all_spk_init(struct snd_soc_pcm_runtime *rtd)
 	struct snd_soc_card *card = rtd->card;
 	int ret;
 
-	ret = snd_soc_dapm_add_routes(&card->dapm, rt1308_speaker_map, 4);
```

Comment:
here,  controls and widgets should be initialized for first_spk_init will not execute.

Reply:
Yeah, that's what I mentioned in the cover, I don't quite understand your code with first, second and all speaker init, so not sure how we can add the widgets/pins once per amp.

It'd be good if you can provide a hint on how you would solve this. Thanks @RanderWang

Reply:
@RanderWang please check sof_sdw_rt1308.c, I tested it on the single amp version, not sure if this is correct.

Reply:
@plbossart  
````
(1) only one speaker in one BE DAI    ---> first_spk_init (BE DAI->init())
(2) two speakers in two BE DAIs       ---> first_spk_init  (BE DAI->init())
                                      ---> second_spk_init (BE DAI->init())
(3) two speakers in one BE DAI        ---> all_spk_init (BE DAI->init())
````

Reply:
Before this PR,  these widget and control are registered by card,  now they are registered in spk_init. so each spk_init should do the job and do it only one time. So second_spk_init doesn't initialized these thing for they are done in first_spk_init. And all_spk_init should do as first_spk_init,
````C
	ret = snd_soc_add_card_controls(card, rt1308_controls,
					ARRAY_SIZE(rt1308_controls));
	if (ret) {
		dev_err(card->dev, "rt1308 controls addition failed: %d\n", ret);
		return ret;
	}

	ret = snd_soc_dapm_new_controls(&card->dapm, rt1308_widgets,
					ARRAY_SIZE(rt1308_widgets));
	if (ret) {
		dev_err(card->dev, "rt1308 widgets addition failed: %d\n", ret);
		return ret;
	}

	ret = snd_soc_dapm_add_routes(&card->dapm, rt1308_map, 4);
	if (ret)
		dev_err(rtd->dev, "failed to add first SPK map: %d\n", ret);
````

Reply:
done but by calling first_spk_init() and second_spk_init in sof_sdw_rt1308.c